client/rest-high-level/src/main/java/org/elasticsearch/client/ml/dataframe/OutlierDetection.java

```diff
@@ -0,0 +1,142 @@
+
+    public static OutlierDetection fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    static final ParseField NAME = new ParseField("outlier_detection");
+    static final ParseField N_NEIGHBORS = new ParseField("n_neighbors");
+    static final ParseField METHOD = new ParseField("method");
+
+    private static ConstructingObjectParser<OutlierDetection, Void> PARSER =
+        new ConstructingObjectParser<>(NAME.getPreferredName(), true,
+            (args) -> {
+                Integer nNeighbors = (Integer) args[0];
+                Method method = (Method) args[1];
+                return new OutlierDetection(nNeighbors, method);
+            });
+
+    private final Integer nNeighbors;
+    private final Method method;
+
+    static {
+        PARSER.declareInt(ConstructingObjectParser.optionalConstructorArg(), N_NEIGHBORS);
+        PARSER.declareField(ConstructingObjectParser.optionalConstructorArg(), p -> {
+            if (p.currentToken() == XContentParser.Token.VALUE_STRING) {
+                return Method.fromString(p.text());
+            }
+            throw new IllegalArgumentException("Unsupported token [" + p.currentToken() + "]");
+        }, METHOD, ObjectParser.ValueType.STRING);
+    }
+
+    /**
+     * Constructs the outlier detection configuration
+     * @param nNeighbors The number of neighbors. Leave unspecified for dynamic detection.
+     * @param method The method. Leave unspecified for a dynamic mixture of methods.
+     */
+    public OutlierDetection(@Nullable Integer nNeighbors, @Nullable Method method) {
+        if (nNeighbors != null && nNeighbors <= 0) {
+            throw new IllegalArgumentException("[" + N_NEIGHBORS.getPreferredName() + "] must be a positive integer");
+        }
+
+        this.nNeighbors = nNeighbors;
+        this.method = method;
+    }
+
+    /**
+     * Constructs the default outlier detection configuration
+     */
+    public OutlierDetection() {
+        this(null, null);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (nNeighbors != null) {
+            builder.field(N_NEIGHBORS.getPreferredName(), nNeighbors);
+        }
+        if (method != null) {
+            builder.field(METHOD.getPreferredName(), method);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        OutlierDetection other = (OutlierDetection) o;
+        return Objects.equals(nNeighbors, other.nNeighbors)
+            && Objects.equals(method, other.method);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(nNeighbors, method);
+    }
+
+    @Override
+    public String getName() { return NAME.getPreferredName(); }
```

Comment:
The method is all on one line. This may be a mischievous IDE setting

Reply:
Done.